tests/compiler.js

```diff
@@ -309,6 +309,14 @@ function testInstantiate(basename, binaryBuffer, name, glue) {
           },
           trace: function(msg, n) {
             console.log("  trace: " + getString(msg) + (n ? " " : "") + Array.prototype.slice.call(arguments, 2, 2 + n).join(", "));
+          },
+          toUpperCaseFromIndex(index, codePointIndex) {
+            const code = String.fromCodePoint(index).toUpperCase().codePointAt(codePointIndex|0);
+            return code !== undefined ? code : -1;
+          },
+          toLowerCaseFromIndex(index, codePointIndex) {
+            const code = String.fromCodePoint(index).toLowerCase().codePointAt(codePointIndex|0);
+            return code !== undefined ? code : -1;
```

Comment:
There can be a `.js` file next to the respective test doing this on a per-test basis, so we don't have to add all of this to the main suite. See for example the `std/mod` test.

Reply:
Good to know! Fixed!